CIP-_/README.md

```diff
@@ -0,0 +1,210 @@
+---
+CIP: ?
+Title: Optimized Coin Selection Algorithms for Cardano
+Category: Wallets
+Authors:
+    - Nicholas Maselli <maselli.nicholas@gmail.com>
+Implementors: CardanoSharp 7, SaturnNFT, Levvy Finance
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/785
+Status: Proposed
+Created: 2024-03-14
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Coin Selection algorithms have become a critical part of the Cardano ecosystem. These algorithms choose a set of input Utxos for a given set of new Outputs.
+
+This CIP aims to present optimized coin selection algorithms to enable Wallets and Dapps to interact with smart contracts and native assets such as tokens and NFTs in a more efficient way. The goal of these algorithms is to primarily reduce the size of transactions as well as reduce the mem and step required for smart contract executions.
+
+## Motivation: why is this CIP necessary?
+
+<b>Cardano needs to scale.</b> As the ecosystem grows and new users come in, blocks are regular at 100% load with 88KB per block allocation. Recently, events such as the Lil Sappys NFT mint have pushed the Cardano blockchain load to over 100% for several hours, preventing anyone from using the chain and providing an awful user experience to new users who complain on social media and give the ecosystem a bad reputation.
```

Comment:
Hi @NicholasMaselli 

Many thanks for creating this CIP!

I have some questions about this event:
> Recently, events such as the Lil Sappys NFT mint have pushed the Cardano blockchain load to over 100% for several hours, preventing anyone from using the chain and providing an awful user experience to new users who complain on social media and give the ecosystem a bad reputation.

Would you be able to give a more complete description of how you believe this particular event is related to the problem of coin selection?

In other words, can you point to specific problems (related to coin selection) that caused or contributed to the situation you describe?

I'd be particularly interested in:
- specific examples of coin selection attempts that failed (for whatever reasons); or
- specific examples of coin selection attempts that produced transactions that were larger than necessary (thus requiring higher fees than necessary)